lib/firehose/server/message_buffer.rb

```diff
@@ -12,27 +12,36 @@ class MessageBuffer
       def initialize(message_list, channel_sequence, consumer_sequence = nil)
         @message_list = message_list
         @channel_sequence = channel_sequence.to_i
-        @consumer_sequence = consumer_sequence.to_i
+
+        if consumer_sequence
```

Comment:
The current behavior when you specify no sequence or a sequence of `0` is to give the latest message. This complexity should not be necessary. Simply calling `to_i` should be fine because they should both give you just the most recent result.